plugins/parquet/parquetembed.hpp

```diff
@@ -211,16 +216,65 @@ class ParquetArrayVisitor : public arrow::ArrayVisitor
         size = 64;
         return arrow::Status::OK();
     }
+    arrow::Status Visit(const arrow::Date32Array &array)
+    {
+        date32_arr = &array;
+        type = DateType;
+        size = 32;
+        return arrow::Status::OK();
+    }
+    arrow::Status Visit(const arrow::Date64Array &array)
+    {
+        date64_arr = &array;
+        type = DateType;
+        size = 64;
+        return arrow::Status::OK();
+    }
+    arrow::Status Visit(const arrow::TimestampArray &array)
+    {
+        timestamp_arr = &array;
+        type = TimestampType;
```

Comment:
is it okay to leave size=0 ?

Reply:
Yes, the size only gets used for the Signed, Unsigned, and Real types to differentiate between the different variations of each that can be stored in parquet.

Reply:
Yes, the size only gets used for the Signed, Unsigned, and Real types to differentiate between the different variations of each that can be stored in parquet.